### PR TITLE
Improved README.md and added Examples

### PR DESCRIPTION
**Pull request by Bibyutatsu**
Hi insilicomedicine,
I have tried updating the Readme file to include more detailed instructions on the installation and running of the model. I have also tried to give some explanation based on my understanding of the model and examples along with some examples of the output images.
Please consider this for the community as I have many colleagues and friends who had hard time running your amazing model.

In the future I am thinking of doing these:
1. Updating the pretrain.ipynb with detailed explanations regarding the steps and significance of the numbers for the different variables.
2. Trying to explain the train_vaelp() and train_rl() functions in detail.
3. Add some explanation regarding how to change the gentrl tokenizer and encoder for custom datasets and tokens.

Thank you
I hope you consider my request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,13 +5,70 @@ The GENTRL model is a variational autoencoder with a rich prior distribution of
...
+## Step 1 :
+Make a new conda environment and install RDKit.
+```
+conda create -c rdkit -n my-rdkit-env rdkit
+```
+Then activate this new environment.
+```
+conda activate my-rdkit-env
+```
+*Note :*  Make sure that the python3 version is 3.5 and higher and pip3 is installed
```
Comment (danpol) [inline README.md]: `or` higher :)

File: README.md [hunk 2]
```diff
@@ -5,13 +5,70 @@ The GENTRL model is a variational autoencoder with a rich prior distribution of
...
-3. Install MOSES from the [repository](https://github.com/molecularsets/moses)
-4. Run the [pretrain.ipynb](./examples/pretrain.ipynb) to train an autoencoder
-5. Run the [train_rl.ipynb](./examples/train_rl.ipynb) to optimize a reward function
+## Step 2 :
+Inside this environment install GENTRL.
+```
+cd <Path_to_GENTRL_folder>
+python3 setup.py install
+```
+*Note :* The `setup.py` doesn't install scikit-learn, so if you don't have it installed please install it using
```
Comment (danpol) [inline README.md]: Feel free to add `scikit-learn` to `setup.py`

File: README.md [hunk 3]
```diff
@@ -5,13 +5,70 @@ The GENTRL model is a variational autoencoder with a rich prior distribution of
...
+```
+pip3 install scikit-learn
+
+or
+
+python3 -m pip install scikit-learn
+```
+
+## Step 3 : (Optional)
+Installing MOSES is optional but if you want to run the examples given [here](https://github.com/Bibyutatsu/GENTRL/tree/master/examples) I would recommend you to install it.
```
Comment (danpol) [inline README.md]: MOSES is in the requirements of `setup.py`, I don't think you should indicate it as an optional step

File: README.md [hunk 4]
```diff
@@ -5,13 +5,70 @@ The GENTRL model is a variational autoencoder with a rich prior distribution of
...
+python3 -m ipykernel install --user --name rdkit_kernel
+```
+Now when you open jupyter notebook. Go to **Change Kernel** > **rdkit_kernel**
+
+With these the installation is over and now we are ready to run the examples provided in the Repo.
+
+# Explanation
+
+## pretrain.ipynb
+In this notebook the Variational Auto Encoder(VAE) is trained. It encodes the SMILES onto the latent space with a rich prior distribution. To put it simply, in this step the structural properties of each training molecule is transferred to a latent space, and process encodings of molecules are created. So, the model learns about the molecules structural properties and interrelation of structure and rewards.
```
Comment (danpol) [inline README.md]:
```suggestion
This notebook trains a Variational Auto Encoder (VAE) with TTLP prior distribution to encode SMILES strings onto the latent space. In this step, VAE maps the structural properties of each training molecule to a latent code.
```

File: README.md [hunk 5]
```diff
@@ -5,13 +5,70 @@ The GENTRL model is a variational autoencoder with a rich prior distribution of
...
+
+
+## train_rl.ipynb
+In this notebook the Reinforecement learning of the model takes place. In this step except the learnable prior all other parameters are kept constant and then the latent space is explored for new molecules with higher rewards. So in this step the model learns to generate better molecules on the basis of reward function.
```
Comment (danpol) [inline README.md]:
```suggestion
This notebook applies reinforcement learning to optimize a reward function. In this step, encoder and decoder are fixed, and the model learns only the learnable prior.
```

File: README.md [hunk 6]
```diff
@@ -5,13 +5,70 @@ The GENTRL model is a variational autoencoder with a rich prior distribution of
...
+
+## sampling.ipynb
+In this notebook new molecules are generated in the form of SMILES. Using RDKit's Draw module a grid of images of these newly generated molecules is produced as shown below ([Link](https://github.com/Bibyutatsu/GENTRL/blob/master/images/Sampling_big.png) of the whole grid).
```
Comment (danpol) [inline README.md]:
```suggestion
This notebook generates new molecules in the form of SMILES strings. Below we show examples of generated molecules (more samples [here](https://github.com/Bibyutatsu/GENTRL/blob/master/images/Sampling_big.png)).
```

**Review by danpol — CHANGES_REQUESTED** [review]
Also, remove changes related to GENTRL pipeline image (both pdf and png), as they seem to be unchanged

File: README.md (hunk 3 above)
Comment (Bibyutatsu) [reply README.md]: Okay, actually in the original steps there were instructions  for installing MOSES dataset. So I added it here but I will remove it.

File: README.md (hunk 2 above)
Comment (Bibyutatsu) [reply README.md]: Yeah will do that.

**Comment by Bibyutatsu** [comment]
Hi, I have made all the changes. Please review it.

**Comment by Bibyutatsu** [comment]
Hi, Can you please review my changes and merge the pull request.

**Comment by gilnahmias** [comment]
Note a missing comma at setup.py, line 14 as per https://youtu.be/ya3AdrfKYzc?t=2470

**Review by Bibyutatsu — COMMENTED** [review]
All the changes incorporated
